standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreListenerNotifier.java

```diff
@@ -261,6 +265,38 @@ public void notify(MetaStoreEventListener listener, ListenerEvent event)
     return event.getParameters();
   }
 
+  /**
+   * Notify a list of listeners about a specific metastore event related to txn. Each listener notified might update
+   * the (ListenerEvent) event by setting a parameter key/value pair. These updated parameters will
+   * be returned to the caller.
+   *
+   * @param listeners List of MetaStoreEventListener listeners.
+   * @param eventType Type of the notification event.
+   * @param event The ListenerEvent with information about the event.
+   * @param dbConn The JDBC connection to the remote meta store db.
+   * @param sqlGenerator The helper class to generate db specific SQL string.
+   * @return A list of key/value pair parameters that the listeners set. The returned object will return an empty
+   *         map if no parameters were updated or if no listeners were notified.
+   * @throws MetaException If an error occurred while calling the listeners.
+   */
+  public static Map<String, String> notifyTxnEvent(List<? extends MetaStoreEventListener> listeners,
```

Comment:
Shall name is more generic to denote directSQL implementation. I would suggest notifyEventWithDirectSql()